### PR TITLE
#306 - Login Page

### PR DESCRIPTION
**Pull request by dleadbetter**
This pull request refactors the Login page according to the designs in this [issue](https://github.com/performant-software/vico/issues/306). The login page will allow for the following branding options:

#### Header bar
An image can be added to `recogito-cloud` to use as a custom header bar image. The image name must be specified in `config.json`.

#### Login logo
An image can be added to `recogito-cloud` to use as a custom login logo. The image name must be specified in `config.json`.

#### Footer component
A custom React component can be added to `recogito` cloud to replace the default footer. The `bottom_logos_enabled` option must be set to "true" in `config.json`. If set to "true" the "powered by Recogito Studio" logo will display below the login buttons.

#### Banner image
A URL for a custom banner image can be specified as the `banner_image` option on `config.json`.

## Default
By default, the "Recogito Studio" logo will display in both the header and footer. Username and password login will be the primary method.

Screenshot 2024-08-26 at 2 32 41 PM

## Branding - Login Logo
A custom branding logo can be configured to display at the top of the login form.

Screenshot 2024-08-26 at 2 34 14 PM

## Branding - Login Logo/SSO First
When SSO is configured as the primary login method, the username and password option will display as an accordion button. Clicking the button will reveal the form.

Screenshot 2024-08-26 at 2 35 51 PM
Screenshot 2024-08-26 at 2 35 59 PM

## Branding - Top Logo
Screenshot 2024-08-26 at 2 38 34 PM

## Branding  - Top Logo/Bottom Logo
Screenshot 2024-08-26 at 2 57 15 PM

## Branding - Top Logo/Bottom Logo/SSO First
Screenshot 2024-08-26 at 2 57 42 PM
Screenshot 2024-08-26 at 2 57 49 PM

## Forgot Password
The `input` elements on the forgot password page have also been updated to match the login page.

Screenshot 2024-08-26 at 2 58 03 PM

## Reset Password
The `input` elements on the reset password page have also been updated to match the login page.

Screenshot 2024-08-26 at 2 58 18 PM

## Join Public Group
The join public group page uses the same `HeaderLayout` as the login page, and has been updated accordingly.

Screenshot 2024-08-26 at 2 58 33 PM

### CODE REVIEW

**Review by lwjameson — APPROVED** [review]
Looks great Derek!

File: src/components/Logo/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
```
Comment (rsimon) [inline src/components/Logo/Logo.tsx:1]: Just a tiny request: could we rename this to "RecogitoLogo" for a more explicit name? BTW: I note that there is a "RecogitoLogo" component currently. But that's an outdated placeholder we can (finally) drop.

**Comment by lwjameson** [comment]
https://github.com/recogito/recogito-client/pull/250#issuecomment-2312548670

Put english in the matching de file as well. @rsimon will translate.

**Comment by rsimon** [comment]
@dleadbetter I can do the DE translation file. Currently, if there is no DE file, labels will show blank in the German UI.

P.S.: if we have time available, we should IMO switch to `i18next` or something similar. (When we started with Astro 2, back in the day, the Astro integration was broken. Hence the hand-rolled workaround. Even today, the [astro-i18next integration](https://github.com/yassinedoghri/astro-i18next/issues) doesn't exactly inspire confidence...)

File: src/i18n/de/branding.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "powered by": "powered by"
```
Comment (rsimon) [inline src/i18n/de/branding.json:2]: The proper German translation (in terms of actual usage in the wild) is indeed... "powered by" ;-)

**Review by rsimon — APPROVED** [review]
This looks great, thanks! I added the one missing German label.

P.S.: just one more tiny request (not crucial, obviously). On the login page, should we have a kind of grey-ish background as a fallback? 

When the page first loaded for me, it looked broken for an instant, because the splash image took some time to load, and the background is the same as the login form:

Bildschirmfoto 2024-08-27 um 15 46 42

With a grey background, it's clear what the page layout is supposed to be.

Bildschirmfoto 2024-08-27 um 15 48 26

**Comment by jamiefolsom** [comment]
> @dleadbetter I can do the DE translation file. Currently, if there is no DE file, labels will show blank in the German UI.
> 
> P.S.: if we have time available, we should IMO switch to `i18next` or something similar. (When we started with Astro 2, back in the day, the Astro integration was broken. Hence the hand-rolled workaround. Even today, the [astro-i18next integration](https://github.com/yassinedoghri/astro-i18next/issues) doesn't exactly inspire confidence...)

We've used https://inlang.com/m/gerre34r/library-inlang-paraglideJs with Astro on other projects; @ajolipa may have some thoughts to share on how that works.

**Pull request merged**